camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WorkflowYamlTest.java

```diff
@@ -1315,4 +1314,79 @@ public void testAddPolicyWithDslFromDeployableBlueprint() throws Exception {
         EntityAsserts.assertAttributeEquals(entity, Sensors.newSensor(Object.class, "result"), "yes");
     }
 
+    @Test
+    public void testSshStepOnLocalhostLocation() throws Exception {
+        String yaml =
+                "location: localhost\n" +
+                "services:\n" +
+                "  - type: " + WorkflowSoftwareProcess.class.getName() +"\n" +
+                "    name: sample-server\n" ;
+        Entity app = createStartWaitAndLogApplication(yaml);
+
+        WorkflowExecutionContext x1 = WorkflowBasicTest.runWorkflow(app.getChildren().iterator().next(), Strings.lines(
+                "lock: x",
+                "steps:",
+                "- type: ssh\n" +
+                "  command: |\n" +
+                "    echo \"init-done\" >> wf.log"), "test");
+        Object result  = x1.getTask(false).get().get();
+        Asserts.assertEquals(((Map)result).get("exit_code"), 0);
+    }
+
+    @Test
+    public void testSshStepOnLocalhostDefinition() throws Exception {
```

Comment:
remove `lock: x`

make integration test - or support mocks for connection definition.  that seems harder, but quite useful.  i think another mechanism we use is to specify the `RecordingSshTool`, so if there is a way to supply a tool class that could work.  (the recording tool does nothing and returns 0 for all ssh commands, but just records the commands.)

Reply:
Done.